Framework/Algorithms/src/CorelliCrossCorrelate.cpp

```diff
@@ -149,11 +149,11 @@ void CorelliCrossCorrelate::exec() {
   std::transform(tdc.begin(), tdc.end(), tdc.begin(), [offset](auto timing) { return timing + offset; });
 
   // Determine period from chopper frequency.
-  auto motorSpeed = dynamic_cast<TimeSeriesProperty<double> *>(inputWS->run().getProperty("BL9:Chop:Skf4:MotorSpeed"));
-  if (!motorSpeed) {
+  auto motorSpeed = inputWS->run().getTimeAveragedValue("BL9:Chop:Skf4:MotorSpeed");
```

Comment:
A different approach would be
```c++
if (!inputWS->run().hasProperty("BL9:Chop:Skf4:MotorSpeed")) {
    throw Exception::NotFoundError("Could not find a log value for the motor speed", "BL9:Chop:Skf4:MotorSpeed");
}
const auto motorSpeed = inputWS->run().getPropertyAsSingleValue("BL9:Chop:Skf4:MotorSpeed");
```
`getPropertyAsSingleValue()` returns the time-average using caching and is not necessary to change, but the way of checking for the property should be.